src/core/public/chrome/chrome_service.tsx

```diff
@@ -152,6 +153,8 @@ export class ChromeService {
     const navLinks = this.navLinks.start({ application, http });
     const recentlyAccessed = await this.recentlyAccessed.start({ http });
     const docTitle = this.docTitle.start({ document: window.document });
+    const { cloud } = npSetup.plugins as any;
+    const isCloudEnabled = !!(cloud && cloud.isCloudEnabled);
```

Comment:
This is going to be an issue for two reasons:

- `'ui/new_platform'` is the trick used to access NP in the legacy world. It's usage in new platform, especially in `core`, should be avoided at all cost.
- `core` should never actively import or depend on plugins.

I'm aware that we were 'cheating' before by using the injectedMetadata (which was **not** an acceptable solution either), and that we need to find something to replace that for the plugin migration, so I've got two propositions:

- Wait until https://github.com/elastic/kibana/pull/51438 lands, and use capabilities to register the  `isCloudEnabled` state. `ChromeService` could then check if `capabilities.cloud.enabled` is true.

- Inverse the control between core and cloud plugin, by exposing a `setCloudEnabled` setter on the `ChromeService`'s setup contract and invoking it in `cloud` plugin's setup phase.

As the first one still explicitly check a plugin's state, even in the capabilities, I think I prefer the second one. Also, this avoid this PR being blocked.

note: after checking the code, I see that the only usage of this property in the header is here: 

https://github.com/elastic/kibana/blob/87e77e96381c8d18202ee7af7fd5836be723f9dd/src/core/public/chrome/ui/header/header_help_menu.tsx#L199-L204

Maybe we want to allow plugin's to register the help/support url instead with a `chrome.setSupportUrl` instead of `chrome.setCloudEnabled`? Seems a little more generic.

CC @joshdover @elastic/kibana-platform  WDYT?

Reply:
As a short term workaround I like `chrome.setSupportUrl` the most.

Though I wonder if Core should be the owner of the content of this menu. It doesn't feel like Core (and the Platform team by extension) should be the owners of how this menu behaves and what it looks like.

Reply:
I've pushed the relevant changes. @rudolf thanks for providing a quick fix for this :heart: